### PR TITLE
fix alias with nullable fields

### PR DESCRIPTION
**Pull request by crissi**
## Summary
Fixes input alias with fields set to null or not set at all

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

### Checklist
- [x] Existing tests have been adapted and/or new tests have been added
- [x] Code style has been fixed via `composer fix-style`

### CODE REVIEW

**Comment by crissi** [comment]
running this though a real life test and discovering some issues

**Comment by mfn** [comment]
Should I still wait for merging or go ahead?

**Comment by crissi** [comment]
should be ready now

File: src/Support/AliasArguments/ArrayKeyChange.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ private function changeKey(array $target, array $segments, string $replaceKey):
         $segment = array_shift($segments);
 
         if (empty($segments)) {
-            if (isset($target[$segment])) {
+            if (array_key_exists($segment, $target) && $replaceKey !== $segment) {
```
Comment (mfn) [inline src/Support/AliasArguments/ArrayKeyChange.php]: I don't see this change fully covered.

When I test it with an `} else { throw new \RuntimeException('boom'); }`, that exception is never thrown.

Is it possible to cover this?

I checked also the other changes, they are 🤞

Comment (crissi) [reply src/Support/AliasArguments/ArrayKeyChange.php]: The null_value field I added covers this it would never, alias if not using array_key_exists

Comment (mfn) [reply src/Support/AliasArguments/ArrayKeyChange.php]: Makes sense, thanks 👍

File: src/Support/AliasArguments/AliasArguments.php [hunk 2]
```diff
@@ -13,9 +13,6 @@
 
 class AliasArguments
 {
-    private $typedArgs;
-    private $arguments;
-
```
Comment (mfn) [inline src/Support/AliasArguments/AliasArguments.php]: Removed them right in this PR as I noticed they were unused after the code was refactored

**Pull request merged**